roles/provision-keycloak-apb/tasks/addrealmuser-keycloak.yml

```diff
@@ -0,0 +1,127 @@
+---
+- name: Generate KC development username
+  shell: tr -d -c "a-zA-Z" < /dev/urandom | head -c 20
+  register: kcdev_username
+
+- name: Generate KC development password
+  shell: tr -d -c "a-zA-Z0-9" < /dev/urandom | head -c 20
+  register: kcdev_password
+
+- name: "Generate keycloak {{ ADMIN_NAME }} auth token"
+  uri:
+    url: "{{ KEYCLOAK_HOST }}/auth/realms/master/protocol/openid-connect/token"
+    method: POST
+    body: "client_id=admin-cli&username={{ ADMIN_NAME }}&password={{ ADMIN_PASSWORD }}&grant_type=password"
+    validate_certs: no
+  register: keycloak_auth_response
+  until: keycloak_auth_response.status == 200
+  retries: 600
+  delay: 2
+
+- set_fact:
+    KCDEV_USERNAME: '{{ kcdev_username.stdout }}'
+    KCDEV_PASSWORD: '{{ kcdev_password.stdout }}'
+
+- name: "Create {{ KCDEV_USERNAME }} user in {{ KEYCLOAK_REALM }} realm"
+  uri:
+    url: "{{ KEYCLOAK_HOST }}/auth/admin/realms/{{ KEYCLOAK_REALM }}/users"
+    method: POST
+    body: '{"enabled":true,"attributes":{},"username":"{{ KCDEV_USERNAME }}","emailVerified":"","email":"{{ KCDEV_USERNAME }}@email.com","firstName":"Openshift","lastName":"Developer"}'
+    validate_certs: no
+    body_format: json
+    headers:
+      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
+    status_code: 201
+  register: kcdev_user_created
+
+- name: "Change {{ KCDEV_USERNAME }} password in {{ KEYCLOAK_REALM }} realm"
+  uri:
+    url: "{{ kcdev_user_created.location }}/reset-password"
+    method: PUT
+    body: '{"type":"password","value":"{{ KCDEV_PASSWORD }}","temporary":false}'
+    validate_certs: no
+    body_format: json
+    headers:
+      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
+    status_code: 204
+
+- name: "Get all available client in {{ KEYCLOAK_REALM }} realm"
+  uri:
+    url: "{{ KEYCLOAK_HOST }}/auth/admin/realms/{{ KEYCLOAK_REALM }}/clients"
+    method: GET
+    validate_certs: no
+    body_format: json
+    headers:
+      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
+    status_code: 200
+  register: kcdev_realm_clients
+
+- set_fact:
+    kcdev_realm_mgmt_client: "{{ kcdev_realm_clients.json | json_query(kcdev_realm_mgmt_jq) | first}}"
+    KCDEV_USERID: "{{ kcdev_user_created.location.split('/')[-1] }}"
+
+- k8s_v1_secret:
+    name: '{{ keycloak_secret_name }}'
+    namespace: '{{ namespace }}'
+    labels:
+      name: keycloak
+      mobile: enabled
+      serviceName: keycloak
+    string_data:
+      admin_username: '{{ ADMIN_NAME }}'
```

Comment:
we shouldn't set this if it is a shared service as it defeats the purpose of creating the new user and realm

Reply:
Done!